### ISSUE TITLE
False/True handled as integer - for what reason?

### ISSUE THREAD

**Issue by GWRon** [bmx-ng/bcc#679]
Take this code:
```
SuperStrict
Framework Brl.StandardIO


Function TakeByte:Int(a:Byte)
End Function


TakeByte(False)
```
it results in:
`Compile Warning: In call to Function TakeByte:Int(a:Byte). Argument #1 is "Int" but declaration is "Byte".`

Is there a reason for "True/False" being internally integers? I know we do not have real booleans for now - but if we consider True is "1" and False is "0" (check with `print (5 * true)` and `print (5 * false)`) then couldn't we use the lowest possible primitive we support? In this case: Byte?

**Comment by GWRon** [comment]
BTW: when not passing False, but the comparison of false, it does not raise an error/warning:
```
TakeByte(False=False)
```

**Comment by GWRon** [comment]
I am not sure why ... but I tried to bisect the bcc changes to identify when it was introduced ... but it works with current master revision and also  older ones.

I then rebuilt bcc from my local folder (did not pull new changes in) and the generated BCCs differ: the new build works (but filesize increased - maybe zlib stuff, updated brl/pub/... mods ...).
Anyways... it works now "as expected" - but I don't know why it did not work with the previous local build.
